### PR TITLE
Add Python 3.11 support

### PR DESCRIPTION
**Pull request by kenodegard**
### Description

Add Python 3.11 to test matrix and minor test updates.

Xref https://github.com/conda/conda-build/issues/4370

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [ ] ~Add / update outdated documentation?~

### CODE REVIEW

**Comment by jezdez** [comment]
pre-commit.ci autofix

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -63,14 +63,14 @@ jobs:
       fail-fast: false
       matrix:
         # test all lower versions (w/ stable conda) and upper version (w/ canary conda)
-        python-version: ['3.8', '3.9']
+        python-version: ['3.8', '3.9', '3.10']
```
Comment (sven6002) [inline .github/workflows/tests.yml:66]: do we still need 3.8.. now..

File: tests/requirements.txt [hunk 2]
```diff
@@ -20,7 +20,6 @@ pkginfo
 psutil
 py  # https://github.com/ContinuumIO/anaconda-issues/issues/13198
 py-lief
-pycrypto
```
Comment (sven6002) [inline tests/requirements.txt:23]: does need to change to https://pypi.org/project/pycryptodome/3.11.0 instead?

File: .github/workflows/tests.yml (hunk 1 above)
Comment (kenodegard) [reply .github/workflows/tests.yml:66]: yes, Python 3.8 support hasn't been dropped yet so it still needs testing

Comment (sven6002) [reply .github/workflows/tests.yml:66]: is the plan to drop it for next release round??

Comment (kenodegard) [reply .github/workflows/tests.yml:66]: at this time we only drop Python 3.8 once it reaches EOL/when support is dropped from the defaults channel

Comment (sven6002) [reply .github/workflows/tests.yml:66]: makes sense

**Pull request merged**